packages/edit-site/src/components/sidebar-dataviews/index.js

```diff
@@ -14,45 +12,13 @@ import { unlock } from '../../lock-unlock';
 import DataViewItem from './dataview-item';
 import CustomDataViewsList from './custom-dataviews-list';
 
-const { useLocation, useHistory } = unlock( routerPrivateApis );
-
-/**
- * Hook to switch to table layout when switching to the trash view.
- * When going out of the trash view, it switches back to the previous layout if
- * there was an automatic switch to table layout.
- */
-function useSwitchToTableOnTrash() {
```

Comment:
This is removed, we don't want specific logic to any default/custom view. They all should work the same, see [slack thread](https://wordpress.slack.com/archives/C02QB2JS7/p1722331514967269) in core-editor channel.